modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java

```diff
@@ -0,0 +1,1122 @@
+			jspWriter.write(" class=\"text-truncate\">");
+
+			jspWriter.write(
+				LanguageUtil.format(
+					resourceBundle, "x-results-for-x",
+					new Object[] {
+						getItemsTotal(),
+						(searchValue == null) ? StringPool.BLANK : searchValue
+					}));
+
+			jspWriter.write("</span></span></div></li>");
+
+			if (filterLabelItems != null) {
+				for (int i = 0; i < filterLabelItems.size(); i++) {
+					_writeLabelItem(
+						jspWriter, filterLabelItems.get(i),
+						i == (filterLabelItems.size() - 1));
+				}
+			}
+
+			jspWriter.write("<li class=\"tbar-item\"><div class=\"");
+			jspWriter.write("tbar-section\">");
+
+			LinkTag linkTag = new LinkTag();
+
+			linkTag.setCssClass("component-link tbar-link");
+			linkTag.setHref(getClearResultsURL());
+			linkTag.setLabel(LanguageUtil.get(resourceBundle, "clear"));
+
+			linkTag.doTag(pageContext);
+
+			jspWriter.write("</div></li></ul></div></nav>");
+		}
+
+		return SKIP_BODY;
+	}
+
+	private Map<String, String> _getParamsMap(String url) {
+		Map<String, String> searchData = new HashMap<>();
+
+		String[] parameters = StringUtil.split(
+			HttpUtil.getQueryString(url), CharPool.AMPERSAND);
+
+		for (String parameter : parameters) {
+			if (parameter.length() == 0) {
+				continue;
+			}
+
+			String[] parameterParts = StringUtil.split(
+				parameter, CharPool.EQUAL);
+
+			if (ArrayUtil.isEmpty(parameterParts)) {
+				continue;
+			}
+
+			String parameterName = parameterParts[0];
+
+			String parameterValue = StringPool.BLANK;
+
+			if (parameterParts.length > 1) {
+				parameterValue = parameterParts[1];
+			}
+
+			parameterValue = HttpUtil.decodeURL(parameterValue);
+
+			searchData.put(parameterName, parameterValue);
+		}
+
+		return searchData;
+	}
+
+	private String _namespace(String namespace, String prop) {
+		if (prop == null) {
+			return null;
+		}
+
+		if (namespace == null) {
+			return prop;
+		}
+
+		return namespace + prop;
+	}
+
+	private void _writeLabelItem(
+			JspWriter jspWriter, LabelItem labelItem, boolean expand)
+		throws Exception {
+
+		jspWriter.write("<li class=\"tbar-item");
+
+		if (expand) {
+			jspWriter.write(" tbar-item-expand");
+		}
+
+		jspWriter.write("\"><div class=\"tbar-section\">");
+
+		LabelTag labelTag = new LabelTag();
+
+		labelTag.setCssClass("component-label tbar-label");
+		labelTag.setDismissible((boolean)labelItem.get("dismissible"));
+		labelTag.setDisplayType("unstyled");
+		labelTag.setLabel((String)labelItem.get("label"));
+
+		labelTag.doTag(pageContext);
+
+		jspWriter.write("</div></li>");
+	}
+
+	private static final String _ATTRIBUTE_NAMESPACE =
+		"clay:management-toolbar:";
+
+	private List<DropdownItem> _actionDropdownItems;
+	private String _checkboxStatus = "unchecked";
+	private String _clearResultsURL;
+	private String _clearSelectionURL;
+	private String _contentRenderer;
+	private CreationMenu _creationMenu;
+	private Boolean _disabled;
+	private ManagementToolbarDisplayContext _displayContext;
```

Comment:
@carloslancha `ManagementToolbarDisplayContext` is used in a few places directly, for example in [ViewFlatUsersDisplayContextFactory](https://github.com/liferay/liferay-portal/blob/master/modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/display/context/ViewFlatUsersDisplayContextFactory.java#L98) . I don't think it is safe to rename it. We could deprecate it and make a new class with the same content and different name, but that is probably not a good idea, renaming should not be a reason for deprecation.

We can rename internally `_displayContext` to `_managementToolbarModel`, but then we would need to change `displayContext` tag attribute. This would then mess up the very widespread pattern (700+ uses) of setting `*displayContext`s, like [AccountUsersAdminManagementToolbarDisplayContext](https://github.com/liferay/liferay-portal/blob/master/modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/display/context/AccountUsersAdminManagementToolbarDisplayContext.java) in [this jsp](https://github.com/liferay/liferay-portal/blob/master/modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_users_admin/view.jsp#L37). These contexts inherit model properties but also have additional stuff. But, when they are passed to tag, we classify them through `ManagementToolbarDisplayContext` interface, ignoring all the rest. 

Liferay Java naming™ :joy:

Reply:
Starting to ~hate~ get a little bit tired about naming xD

Not sure what to answer here to be honest. I'm going to invoke @jbalsas to see if he can throw some light on what is expected here. Thx Marko!

Reply:
I'm going to be neutral about this: 

I both understand @carloslancha desire to rename this to `ManagementToolbar` to be consistent with our Card tags but I also understand @markocikos comment about the fact that since it's already being used in lots of places we should keep the same name. Sorry guys.

Reply:
Not having too much context, I would defer any renaming until Brian asks for it. Specially if it's an already used API